File: baekjoon/구현/1316_그룹단어 체커.js
const fs = require('fs')
const [n, ...input] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split('\n')

function solution() {
  let answer = input.length

  // 단어의 수 만큼 순회
  input.forEach((word) => {
    const check = []
    // 문자열 각 요소의 문자 순회
    for (let i = 0; i < word.length; i++) {
      // 문자가 배열에 있고 앞에 문자열과 다른 경우 -> 그룹 단어 x
      if (check.indexOf(word[i]) !== -1 && word[i] !== word[i - 1]) {
        answer -= 1
        break
      } else {
        check.push(word[i])
      }
    }
  })

  return answer
}

console.log(solution())

File: baekjoon/구현/1316_그룹단어 체커.py
import sys
input = sys.stdin.readline
n = int(input())
input = [input().strip() for _ in range(n)]


def solution():
    answer = len(input)

    # 단어의 수 만큼 순회
    for word in input:
        wordList = list(word) # 문자열 쪼개기
        check = {}
        # 쪼갠 문자열 순회
        for i in range(len(wordList)):
            # 문자가 딕셔너리에 존재할 시
            if wordList[i] in check:
                number = check.get(wordList[i])
                # 현재 인덱스 -1이 딕셔너리에 존재하는 값이라면 연속해서 나타나서 그룹 단어이다.
                if i-1 == number:
                    check[wordList[i]] = i
                # 인덱스가 안 맞다면 그룹 단어가 아니다.
                else:
                    answer -= 1
                    break
            # 객체에 없는 경우 key:현재값, value:현재 인덱스
            else:
                check[wordList[i]] = i

    return answer


print(solution())
